server_installation/topics/operating-mode/crossdc/proc-configuring-infinispan.adoc

```diff
@@ -0,0 +1,137 @@
+
+.Prerequisites
+
+* Install and set up {jdgserver_name} Server.
+
+.Procedure
+
+. Open `infinispan.xml` for editing.
++
+By default, {jdgserver_name} Server uses `server/conf/infinispan.xml` for static configuration such as cluster transport and security mechanisms.
+
+. Create a stack that uses TCPPING as the cluster discovery protocol.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<stack name="global-cluster" extends="tcp">
+    <!-- Remove MPING protocol from the stack and add TCPPING -->
+    <TCPPING initial_hosts="server1[7800],server2[7800]" <1>
+             stack.combine="REPLACE" stack.position="MPING"/>
+</stack>
+----
+<1> Lists the host names for `server1` and `server2`.
++
+. Configure the {jdgserver_name} cluster transport to perform Cross-Datacenter replication.
+.. Add the RELAY2 protocol to a JGroups stack.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<jgroups>
+   <stack name="xsite" extends="udp"> <1>
+      <relay.RELAY2 site="site1" <2>
+                    max_site_masters="1000"/> <3>
+      <remote-sites default-stack="global-cluster"> <4>
+         <remote-site name="site1"/>
+         <remote-site name="site2"/>
+      </remote-sites>
+   </stack>
+</jgroups>
+----
+<1> Creates a stack named `xsite` that extends the default UDP cluster transport.
+<2> Adds the RELAY2 protocol and names the cluster you are configuring as `site1`. The site name must be unique to each {jdgserver_name} cluster.
+<3> Sets 1000 as the number of relay nodes for the cluster. You should set a value that is equal to or greater than the maximum number of nodes in your {jdgserver_name} cluster.
+<4> Names all {jdgserver_name} clusters that backup caches with {jdgserver_name} data and uses the default TCP stack for inter-cluster transport.
++
+.. Configure the {jdgserver_name} cluster transport to use the stack.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<cache-container name="default" statistics="true">
+      <transport cluster="${infinispan.cluster.name:cluster}"
+                 stack="xsite"/> <1>
+</cache-container>
+----
+<1> Uses the `xsite` stack for the cluster.
++
+. Configure the keystore as an SSL identity in the server security realm.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<server-identities>
+  <ssl>
+    <keystore path="server.jks" <1>
+              relative-to="infinispan.server.config.path"
+              keystore-password="password" <2>
+              alias="server" /> <3>
+  </ssl>
+</server-identities>
+----
+<1> Specifies the path of the keystore that contains the SSL identity.
+<2> Specifies the password to access the keystore.
+<3> Names the alias of the certificate in the keystore.
++
+. Configure the authentication mechanism for the Hot Rod endpoint.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<endpoints socket-binding="default">
+   <hotrod-connector name="hotrod">
+      <authentication>
+         <sasl mechanisms="SCRAM-SHA-512" <1>
```

Comment:
```suggestion
         <sasl mechanisms="DIGEST-MD5" <1>
```

Reply:
@andymunro I apologise for the churn here because I believe I requested this change initially. As it turns out, the Hot Rod client in SSO is Infinispan 9.4.x, not Infinispan 11.x. I wasn't aware of that but SCRAM-SHA-512 wasn't added until Infinispan 10 so this probably should be DIGEST-MD5.

Reply:
@oraNod Note: I've also tested this with SCRAM-SHA-512 and it worked.

Reply:
@tkyjovsk I changed it back to DIGEST-MD5 and I am ready to commit. which one is better?

Reply:
MD5 is considered weak and insecure. SHA-512 should be stronger and more secure. I think it should be supported. I couldn't find it in the Infinispan 9 docs however I can see it mentioned in [Wildfly 17 docs](https://docs.wildfly.org/17/WildFly_Elytron_Security.html#Passwords) already, and I can see `wildfly-elytron-sasl-scram` library in curren KC/SSO version, so I would say it's supported. Unless there are objections, I would go with SCRAM-SHA-512. @mposolda @oraNod

Reply:
@tkyjovsk I think it depends on the version of the Hot Rod client in SSO. I was using Keycloak 12 which has "infinispan-client-hotrod-11.0.4.Final.jar" which supports SCRAM-SHA-512 and is the default. Client versions at Infinispan 9.4 and earlier would need to use DIGEST-MD5. Maybe it comes down to which Keycloak/SSO versions we support in combo with Infinispan 11/Data Grid 8. 

@andymunro Once again I regret the back and forth on this one but if we document a SASL mech that the client doesn't support then customers won't be able to get the integration working. I think we need to be clear about which versions can use which mechanisms.

Reply:
Thanks, @oraNod for all the input.  I think for now the consensus is SCRAM-SHA=512, but we may change it in the future.

Reply:
I think that `SCRAM-SHA-512` probably works with latest Keycloak master, which uses infinispan 11. However I think that it won't work with RHSSO 7.4.X, which still uses infinispan 9. I think RHSSO 7.4 will need to use DIGEST-MD5 (will be good to doublecheck). I suppose documentation will need to cover this and there will probably be difference between Keycloak master and RHSSO 7.4 docs.

Reply:
@mposolda, @tkyjovsk  I made SCRAM-SHA-512 appear conditionally in upstream and DIGEST-MD5 in downstream in all locations for the new release of Data Grid and Infinispan

If you can confirm DIGEST-MD5 for RH SSO, I think we are done.